File: src/main/java/ru/lewis/libraryloader/TomlParser.java
package ru.lewis.libraryloader;

import ru.lewis.libraryloader.model.Dependency;

import java.util.*;

public class TomlParser {

    public record LibrariesConfig(Map<String, String> repositories, List<Dependency> libraries) {
    }

    public static LibrariesConfig parse(String content) {
        Map<String, String> repositories = new LinkedHashMap<>();
        List<Dependency>    libraries    = new ArrayList<>();
        String section = "";

        for (String raw : content.split("\n")) {
            String line = raw.trim();
            if (line.isBlank() || line.startsWith("#")) continue;

            if (line.equals("[repositories]")) { section = "repositories"; continue; }
            if (line.equals("[libraries]"))    { section = "libraries";    continue; }

            if (line.contains("=")) {
                String[] parts = line.split("=", 2);
                String key   = parts[0].trim().replace("\"", "");
                String value = parts[1].trim().replace("\"", "");

                switch (section) {
                    case "repositories" -> repositories.put(key, value);
                    case "libraries"    -> libraries.add(Dependency.of(value));
                }
            }
        }

        return new LibrariesConfig(repositories, libraries);
    }
}

File: src/main/kotlin/ru/lewis/libraryloader/TomlParser.kt
package ru.lewis.libraryloader

import ru.lewis.libraryloader.model.Dependency

object TomlParser {

    data class LibrariesConfig(
        val repositories: Map<String, String>,
        val libraries: List<Dependency>
    )

    fun parse(content: String): LibrariesConfig {
        val repositories = mutableMapOf<String, String>()
        val libraries = mutableListOf<Dependency>()

        var section = ""

        content.lines()
            .map { it.trim() }
            .filter { it.isNotBlank() && !it.startsWith("#") }
            .forEach { line ->
                when {
                    line == "[repositories]" -> section = "repositories"
                    line == "[libraries]"    -> section = "libraries"
                    line.contains("=") -> {
                        val (key, value) = line.split("=", limit = 2)
                            .map { it.trim().trim('"') }

                        when (section) {
                            "repositories" -> repositories[key] = value
                            "libraries"    -> libraries.add(Dependency.of(value))
                        }
                    }
                }
            }

        return LibrariesConfig(repositories, libraries)
    }
}